code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -153,59 +186,61 @@
 	var/chosen_target = pick(Targets)//Pick the remaining targets (if any) at random
 	return chosen_target
 
-/mob/living/simple_animal/hostile/CanAttack(var/atom/the_target)//Can we actually attack a possible target?
-	if(!the_target)
-		return 0
-	if(see_invisible < the_target.invisibility)//Target's invisible to us, forget it
-		return 0
+// Please do not add one-off mob AIs here, but override this function for your mob
+/mob/living/simple_animal/hostile/CanAttack(atom/the_target)//Can we actually attack a possible target?
+	if(isturf(the_target) || !the_target || the_target.type == /atom/movable/lighting_object) // bail out on invalids
+		return FALSE
+
+	if(ismob(the_target)) //Target is in godmode, ignore it.
+		var/mob/M = the_target
+		if(M.status_flags & GODMODE)
+			return FALSE
+
+	if(see_invisible < the_target.invisibility) //Target's invisible to us, forget it
+		return FALSE
 	if(search_objects < 2)
 		if(isliving(the_target))
 			var/mob/living/L = the_target
-			var/faction_check = faction_check(L)
+			var/faction_check = faction_check_mob(L)

```

Comment:
our version of faction_check_mob uses \ref, is it supposed to use UID? could that be why the check is not working?
https://github.com/ParadiseSS13/Paradise/blob/eb75c8bd398eb8975155cec25615dc47fa48c701/code/modules/mob/mob.dm#L1336